File: archived/v1/src/python/ch01/1.3/l0/demo_properties.py
from parity import is_even, is_odd, parity_bit
from divisibility import is_divisible
from remainder import div_identity, week_shift

def main():
    print("Parity:")
    for n in [10, 7]:
        print(f"  {n} is_even? {is_even(n)}  is_odd? {is_odd(n)}  (bit) {parity_bit(n)}")

    print("\nDivisibility:")
    print("  12 divisible by 3?", is_divisible(12, 3))
    print("  14 divisible by 5?", is_divisible(14, 5))

    print("\nRemainders & identity:")
    q, r = div_identity(17, 5)
    print(f"  17 = 5*{q} + {r}")

    print("\nClock (7-day week):")
    # 0=Mon, 5=Sat
    print("  Saturday(5) + 10 days ->", week_shift(5, 10))

if __name__ == "__main__":
    main()

File: archived/v1/src/go/ch01/1.3/l0/demo_properties.go
package main

import "fmt"

func main() {
	fmt.Println("Parity:")
	for _, n := range []int{10, 7} {
		fmt.Printf("  %d is_even? %v  is_odd? %v  (bit) %s\n",
			n, IsEven(n), IsOdd(n), ParityBit(n))
	}

	fmt.Println("\nDivisibility:")
	fmt.Printf("  12 divisible by 3? %v\n", IsDivisible(12, 3))
	fmt.Printf("  14 divisible by 5? %v\n", IsDivisible(14, 5))

	fmt.Println("\nRemainders & identity:")
	q, r := DivIdentity(17, 5)
	fmt.Printf("  17 = 5*%d + %d\n", q, r)

	fmt.Println("\nClock (7-day week):")
	// 0=Mon, 5=Sat; Sat + 10 -> Mon(1)
	fmt.Println("  Saturday(5) + 10 days ->", WeekShift(5, 10))

	fmt.Println("\nLast digit of 2^15:")
	fmt.Println("  ", PowMod(2, 15, 10))
}
